src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
+    auto parentIt = _performanceTraces.find(parentId);
+    if (parentIt == _performanceTraces.end()) {
+        assert(false && "Parent transaction/span does not exist");
+        return 0;
+    }
+
+    PerformanceTrace &parent = parentIt->second;
+    sentry_span_t *span = nullptr;
+    if (parent.isSpan()) {
+        if (!parent.span()) {
+            assert(false && "Parent span is not valid");
+            return 0;
+        }
+        span = sentry_span_start_child(parent.span(), name.c_str(), description.c_str());
+        assert(span);
+    } else {
+        if (!parent.transaction()) {
+            assert(false && "Parent transaction is not valid");
+            return 0;
+        }
+        span = sentry_transaction_start_child(parent.transaction(), name.c_str(), description.c_str());
+        assert(span);
+    }
+
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
+    if (!res) {
+        assert(false && "Transaction already exists");
+        return 0;
+    }
+
+    PerformanceTrace &pTrace = it->second;
+    pTrace.setSpan(span);
+    parent.addChild(pTrace.id());
+    return traceId;
+}
+
+pTraceId Handler::startPTrace(PTraceName pTraceName, int syncDbId) {
+    std::scoped_lock lock(_mutex);
+    if (!_isSentryActivated || pTraceName == PTraceName::None) return 0;
+
+    pTraceId newPTraceId = 0;
+    if (PTraceInfo pTraceInfo(pTraceName); pTraceInfo._parentPTraceName != PTraceName::None) {
+        // Fetch the pTraceMap associated with the provided syncDbId
+        auto pTraceMapIt = _pTraceNameToPTraceIdMap.find(syncDbId);
+        if (pTraceMapIt == _pTraceNameToPTraceIdMap.end()) {
+            assert(false && "Parent transaction/span is not running.");
+            return 0;
+        }
+        auto pTraceMap = pTraceMapIt->second;
+
+        // Find the parent pTrace
+        auto parentPTraceIt = pTraceMap.find(pTraceInfo._parentPTraceName);
+        if (parentPTraceIt == pTraceMap.end() || parentPTraceIt->second == 0) {
+            assert(false && "Parent transaction/span is not running.");
+            return 0;
+        }
+
+        const pTraceId &parentId = parentPTraceIt->second;
+
+        // Start the span
+        newPTraceId = startSpan(pTraceInfo._pTraceTitle, pTraceInfo._pTraceDescription, parentId);
+        _pTraceNameToPTraceIdMap[syncDbId][pTraceInfo._pTraceName] = newPTraceId;
+    } else {
+        // Start the transaction
+        newPTraceId = startTransaction(pTraceInfo._pTraceTitle, pTraceInfo._pTraceDescription);
+        _pTraceNameToPTraceIdMap[syncDbId][pTraceInfo._pTraceName] = newPTraceId;
+    }
+    return newPTraceId;
+}
+
+void Handler::stopPTrace(PTraceName pTraceName, int syncDbId, bool aborted) {
+    std::scoped_lock lock(_mutex);
+    if (!_isSentryActivated || pTraceName == PTraceName::None) return;
```

Comment:
Can this be moved at the very beginning?

Reply:
Done here: [dc43287](https://github.com/Infomaniak/desktop-kDrive/pull/403/commits/dc432879d2bb5f405a700f650f96ff037efa30f2)